builder/codebuild/codebuild.go

```diff
@@ -0,0 +1,245 @@
+// container.
+type Builder struct {
+	codebuild *codebuild.CodeBuild
+
+	// Required field, arn of the instance-role required by codebuild
+	ServiceRole string
+
+	// The Image used by codebuild to build images. Defaults to 
+	// DefaultCodebuildImage
+	Image string
+
+	// The computing instances AWS CodeBuild will use. Defaults to
+	// DefaultCodebuildComputeType
+	ComputeType string
+
+	// Credentials for Dockerhub
+	DockerUsername string
+	DockerPassword string
+}
+
+type BuildSpecInput struct {
+	// Extend all the values already given in builder
+	*Builder
+
+	// The repository which is being built
+	Repository string
+
+	// The commit sha at which to do the build
+	Sha string
+}
+
+// NewBuilder returns a new Builder backed by the codebuild client.
+func NewBuilder(config client.ConfigProvider) *Builder {
+	return &Builder{
+		codebuild: codebuild.New(config),
+	}
+}
+
+// NewBuilderFromEnv returns a new Builder with a codebuild client
+// configured from the standard Docker environment variables.
+func NewBuilderFromEnv() (*Builder, error) {	
+
+	serviceRole := os.Getenv("CODEBUILD_SERVICE_ROLE")
+
+	if serviceRole == "" {
+		return nil, errors.New("CODEBUILD_SERVICE_ROLE must be set when using codebuild builder")
+	}
+
+	image := os.Getenv("CODEBUILD_IMAGE")
+
+	if image == "" {
+		image = DefaultCodebuildImage
+	}
+
+	computeType := os.Getenv("CODEBUILD_COMPUTE_TYPE")
+
+	if computeType == "" {
+		computeType = DefaultCodebuildComputeType
+	}
+
+	dockerUsername := os.Getenv("DOCKER_USERNAME")
+	dockerPassword := os.Getenv("DOCKER_PASSWORD")
+
+	if dockerUsername == "" || dockerPassword == "" {
+		return nil, errors.New("DOCKER_USERNAME and DOCKER_PASSWORD env vars must be set when using codebuild builder")
+	}
+
+	sess := session.Must(session.NewSession())
+
+	return &Builder{
+		codebuild: codebuild.New(sess),
+		ServiceRole: serviceRole,
+		Image: image,
+		ComputeType: computeType,
+		DockerUsername: dockerUsername,
+		DockerPassword: dockerPassword,
+	}, nil
+}
+
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+// Build executes the docker image.
+func (b *Builder) Build(ctx context.Context, w io.Writer, opts builder.BuildOptions) (image string, err error) {
+	image = fmt.Sprintf("%s:%s", opts.Repository, opts.Sha)
+	err = b.build(ctx, w, opts)
+	return
+}
+
+// Build executes the codebuild image.
+func (b *Builder) build(ctx context.Context, w io.Writer, opts builder.BuildOptions) error {
+	
+	projectName := strings.Join([]string{
+		"conveyor",
+		strings.Replace(opts.Repository, "/", "_", -1),
+	}, "-")
+
+	startBuild, err := b.startBuild(opts, projectName)
+
+	if err != nil {
+
+		awsErr, ok := err.(awserr.Error)
+
+    	if ok && awsErr.Code() == "ResourceNotFoundException" {
+
+	        createResp, err := b.createProject(opts, projectName)
+			check(err)
+			log.Printf("%v", createResp)  
+
+			startBuild, err = b.startBuild(opts, projectName)
+			check(err)
+	        
+	    } else {
+
+	    	return err
+	   	}
```

Comment:
Do yourself a favor and get [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports) running in file save.